htdocs/core/modules/barcode/mod_barcode_product_standard.php

```diff
@@ -224,18 +224,22 @@ public function getNextValue($objproduct = null, $type = '')
 	 *
 	 *	@param	DoliDB		$db					Database handler
 	 *	@param	string		$code				Code to check/correct
-	 *	@param	Product		$product			Object product
-	 *  @param  int		  	$thirdparty_type   	0 = customer/prospect , 1 = supplier
+	 *	@param	Product|Societe	$product			Object product
```

Comment:
I see no reason to allow "Societe" here in documentation. In this file method should be called only for Product or Service.
Why allowing Societe here ?

Reply:
We have mod_barcode_product.... and mod_barcode_societe... deriving from the same parent class.  The method signatures need to be the same.

Another class could be created, one for barcode_product and one for barcode_thirdparty, but I am focussing on applying typing hints for what they are rather than adding structural changes to fix the notificitions.

In general it would be better to test the class type (instanceof) rather than relying solely on the type/element text field or filename.